### ISSUE TITLE
[bug] Enviroment Variables are not expanded when using Sage

### ISSUE THREAD

**Issue by sinsimon** [roots/bud#1724]
### Agreement

- [X] This is not a duplicate of an existing issue
- [X] I have read the [guidelines for Contributing to Roots Projects](https://github.com/roots/.github/blob/master/CONTRIBUTING.md)
- [X] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/) community

### Describe the issue

When using sage 10 variables are read from bedrock .env file, trying to find WP_HOME. 
When using the command` bud build` and a .env file using the expanded format it throws an error.
```
$ bud build
⠹ 
[@roots/bud-preset-wordpress] › ✖  error on init 
 TypeError [ERR_INVALID_URL]: Invalid URL
    at new NodeError (node:internal/errors:371:5)
    at onParseError (node:internal/url:552:9)
    at parse (<anonymous>)
    at new URL (node:internal/url:628:5)
    at Object.set origin [as origin] (file:///Users/simonem/sites/catapano/web/app/themes/sage/node_modules/@roots/bud-preset-wordpress/lib/extension.js:11:57)
    at Object.init (file:///Users/simonem/sites/catapano/web/app/themes/sage/node_modules/@roots/bud-preset-wordpress/lib/extension.js:18:21)
    at Object._init (file:///Users/simonem/sites/catapano/web/app/themes/sage/node_modules/@roots/bud-framework/lib/extension/index.js:67:24)
    at async Extensions.run (file:///Users/simonem/sites/catapano/web/app/themes/sage/node_modules/@roots/bud-extensions/lib/service.js:194:13)
    at async file:///Users/simonem/sites/catapano/web/app/themes/sage/node_modules/@roots/bud-extensions/lib/service.js:222:25
    at async Extensions.runDependencies (file:///Users/simonem/sites/catapano/web/app/themes/sage/node_modules/@roots/bud-extensions/lib/service.js:214:13) {
  input: '${APP_NAME}.test',
  code: 'ERR_INVALID_URL'
[bud@6.3.3] [bud] › ✖  error in @roots/bud-preset-wordpress
✨  Done in 4.95s.
```
Here's my .env file
```
APP_NAME=catapano 
WP_HOME=${APP_NAME}.test
```
This happens also if I set the sage .proxy() option manually like this on my bud.config.js
```
export default async(app) => {
    app
    /**
     * Application entrypoints
     */
        .entry({
        app: ["@scripts/app", "@styles/app"],
        editor: ["@scripts/editor", "@styles/editor"],
    })

    /**
     * Directory contents to be included in the compilation
     */
    .assets(["images"])

    /**
     * Matched files trigger a page reload when modified
     */
    .watch(["resources/views/**/*", "app/**/*"])

    /**
     * Proxy origin (`WP_HOME`)
     */
    .proxy("https://catapano.test/")

```

### Expected Behavior

Bud expands variables like this https://www.npmjs.com/package/dotenv-expand or ignores non-correct formats without throwing errors.

### Actual Behavior

Bud thows errors on bud build when having a WP_HOME on bedrock .env with an incorrect url format

### Steps To Reproduce

1. add a non standard url format on bedrock .env with sage installed
2. run bud build on sage folder

### version

6.3.3

### What package manager are you using?

yarn

### version

1.22.4

**Comment by kellymears** [comment]
It causes too many unexpected issues to try and auto-expand `.env` values from outside of the project root. for instance, wordpress salts often need to be escaped. I wouldn't feel comfortable expanding lower level .env until that is handled reliably.

there is actually a unit test enforcing it 

https://github.com/roots/bud/blob/4b7b621a6711f660efc44cbdd27ddaf150ae5862/tests/unit/bud/context/env.test.ts#L60-L64

**Comment by sinsimon** [comment]
@kellymears Got it, thanks. Is there a way to have WP_HOME interpolated and have bud ignore it? Is there a config value I can set? I tried .proxy and it didn't work.

**Comment by kellymears** [comment]
#1727 should provide a better way to bail on this but for now I would set `WP_HOME` in a `.env` _in theme root_. The project .env gets final say and it should override the one in bedrock.

**Comment by kellymears** [comment]
I think this is fixed in 6.4.5. See https://bud.js.org/releases/6.4.5/#-rootsbud-preset-wordpress-improve-handling-of-wp_home.
